heed/src/env.rs

```diff
@@ -29,34 +30,25 @@ use crate::{
 
 /// The list of opened environments, the value is an optional environment, it is None
 /// when someone asks to close the environment, closing is a two-phase step, to make sure
-/// noone tries to open the same environment between these two phases.
+/// none tries to open the same environment between these two phases.
 ///
-/// Trying to open a None marked environment returns an error to the user trying to open it.
-static OPENED_ENV: Lazy<RwLock<HashMap<PathBuf, EnvEntry>>> = Lazy::new(RwLock::default);
+/// Trying to open a `None` marked environment returns an error to the user trying to open it.
+///
+/// [same_file::Handle] abstract the platform details of checking if the given paths points to
+///  the same file on the filesystem.
+///
+/// ## Safety
+///
+/// Mind that Handle currently open the file, so avoid writing through the fd held by [same_file::Handle],
+/// since the file will also be opened by LMDB.
+static OPENED_ENV: Lazy<RwLock<HashMap<Handle, EnvEntry>>> = Lazy::new(RwLock::default);
```

Comment:
Ok, so can we add a comment about it? I mean a really documentation comment in the lib.rs file. Under an Important/Remark header, please? I don't know how LMDB behaves towards that.

I am not sure that we really care [about this sentence](http://www.lmdb.tech/doc/) as we keep it open the whole env-lifetime right?
> Do not have open an LMDB database twice in the same process at the same time. Not even from a plain open() call - close()ing it breaks flock() advisory locking.

Reply:
Yeah my worries comes from that line, and my first idea to just check `(dev_t, ino_t`)

For sure it may break the `flock()`ing (not the best system api..), and the multi-process features of lmdb is really handy.

I will try to checkout lmdb sources.

Reply:
So, what do we conclude? Can we still use the `same_file::Handle` struct? If not, what do we plan? Is it even possible to get the `dev_t` and `ino_t` without opening the file?

Reply:
`stat` function family (except the `fstatat`) check metadata and return and don't maintain a file descriptor at least on unix.

It's the kind of function used by `metadata` in rust std, sadly posix never standardized a "device / FSID" so ino_t/dev_t is left to implementation detail.